pkg/agent/core/ngt/handler/grpc/index_test.go

```diff
@@ -380,88 +388,503 @@ func Test_server_IndexInfo(t *testing.T) {
+					Stored:      insertCnt,
+					Uncommitted: removeCnt,
+					Indexing:    false,
+					Saving:      false,
+				},
+			},
+		},
+		{
+			name: "Equivalence Class Testing case 2.4: return uncommitted count with 100 uncommitted insert+delete index",
+			args: args{
+				in1: &payload.Empty{},
+			},
+			fields: fields{
+				name:    name,
+				ip:      ip,
+				svcCfg:  defaultSvcCfg,
+				svcOpts: defaultSvcOpts,
+			},
+			beforeFunc: func(t *testing.T, a args, s *server) {
+				ctx := context.Background()
+
+				// we need to insert request first before remove
+				totalInsertCnt := insertCnt + removeCnt
+				req, err := request.GenMultiInsertReq(request.Float, vector.Gaussian, totalInsertCnt, dim, defaultInsertConfig)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := s.MultiInsert(ctx, req); err != nil {
+					t.Fatal(err)
+				}
+				if _, err := s.CreateIndex(ctx, &payload.Control_CreateIndexRequest{
+					PoolSize: uint32(len(req.Requests)),
+				}); err != nil {
+					t.Fatal(err)
+				}
+
+				// remove
+				if _, err := s.MultiRemove(ctx, request.GenMultiRemoveReq(removeCnt, defaultRemoveConfig)); err != nil {
+					t.Fatal(err)
+				}
+				// insert again
+				req, err = request.GenMultiInsertReq(request.Float, vector.Gaussian, insertCnt, dim, defaultInsertConfig)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := s.MultiInsert(ctx, req); err != nil {
+					t.Fatal(err)
+				}
+			},
+			want: want{
+				wantRes: &payload.Info_Index_Count{
+					Stored:      insertCnt + removeCnt,
+					Uncommitted: insertCnt + removeCnt,
+					Indexing:    false,
+					Saving:      false,
+				},
+			},
+		},
+		{
+			name: "Equivalence Class Testing case 3.1: return when NGT is indexing",
+			args: args{
+				in1: &payload.Empty{},
+			},
+			fields: fields{
+				name: name,
+				ip:   ip,
+				svcCfg: &config.NGT{
+					Dimension:    dim,
+					DistanceType: ngt.Angle.String(),
+					ObjectType:   ngt.Float.String(),
+					KVSDB: &config.KVSDB{
+						Concurrency: 1,
+					},
+					VQueue: &config.VQueue{
+						InsertBufferPoolSize: 10000,
+						DeleteBufferPoolSize: 10000,
+					},
+				},
+				svcOpts: append(defaultSvcOpts,
+					service.WithAutoIndexLength(10000),
+					service.WithAutoIndexCheckDuration("10s"),
+					service.WithAutoSaveIndexDuration("10s"),
+					service.WithAutoIndexDurationLimit("1s"),
+					service.WithDefaultPoolSize(100),
+				),
+			},
+			beforeFunc: func(t *testing.T, a args, s *server) {
+				ctx := context.Background()
+
+				insertCnt := 5000
+				req, err := request.GenMultiInsertReq(request.Float, vector.Gaussian, insertCnt, dim, defaultInsertConfig)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := s.MultiInsert(ctx, req); err != nil {
+					t.Fatal(err)
+				}
+				go func() {
+					if _, err := s.CreateIndex(ctx, &payload.Control_CreateIndexRequest{
+						PoolSize: uint32(insertCnt),
+					}); err != nil {
+						t.Fatal(err)
+					}
+				}()
```

Comment:
I suppose it is not needed anymore.

Reply:
It seems that it is needed, otherwise the create index operation will not be executed automatically.

Even I set the `AutoIndexCheckDuration` to `1s` and `AutoIndexLength` to 1000 with 5000 insert count, the index operation doesn't triggered automatically.
Related logic: https://github.com/vdaas/vald/blob/master/pkg/agent/core/ngt/service/ngt.go#L548

Is there something I was missing? 🤔